Source/OxyPlot/Series/ExtrapolationLineSeries.cs

```diff
@@ -0,0 +1,322 @@
+        public bool IgnoreExtraplotationForScaling { get; set; }
+
+        /// <summary>
+        /// Gets the list of X intervals within which the line is rendered using the second color and style.
+        /// </summary>
+        public List<DataRange> Intervals { get; } = new List<DataRange>();
+
+        /// <summary>
+        /// Gets the actual dash array for the extrapolated parts of the line.
+        /// </summary>
+        protected double[] ActualExtrapolationDashArray
+        {
+            get
+            {
+                return this.Dashes ?? this.ActualExtrapolationLineStyle.GetDashArray();
+            }
+        }
+
+        /// <summary>
+        ///  Overridden to show solid style.Renders the legend symbol for the line series on the
+        /// specified rendering context.
+        /// </summary>
+        /// <param name="rc">The rendering context.</param>
+        /// <param name="legendBox">The bounding rectangle of the legend box.</param>
+        public override void RenderLegend(IRenderContext rc, OxyRect legendBox)
+        {
+            double xmid = (legendBox.Left + legendBox.Right) / 2;
+            double ymid = (legendBox.Top + legendBox.Bottom) / 2;
+
+            var pts = new[] { new ScreenPoint(legendBox.Left, ymid), new ScreenPoint(xmid, ymid) };
+
+            rc.DrawLine(
+                pts,
+                this.GetSelectableColor(this.ActualColor),
+                this.StrokeThickness,
+                this.EdgeRenderingMode,
+                this.ActualDashArray);
+
+            pts = new[] { new ScreenPoint(xmid, ymid), new ScreenPoint(legendBox.Right, ymid) };
+
+            rc.DrawLine(
+                pts,
+                this.GetSelectableColor(this.ActualExtrapolationColor),
+                this.StrokeThickness,
+                this.EdgeRenderingMode,
+                this.ActualExtrapolationDashArray);
+
+            var midpt = new ScreenPoint(xmid, ymid);
+
+            rc.DrawMarker(
+                midpt,
+                this.MarkerType,
+                this.MarkerOutline,
+                this.MarkerSize,
+                this.ActualMarkerFill,
+                this.MarkerStroke,
+                this.MarkerStrokeThickness,
+                this.EdgeRenderingMode);
+        }
+
+        /// <summary>
+        /// Updates the data and sorts the intervals.
+        /// </summary>
+        protected internal override void UpdateData()
+        {
+            base.UpdateData();
+
+            this.orderedIntervals = this.MergeOverlaps(this.Intervals);
+        }
+
+        /// <summary>
+        /// Updates the maximum and minimum values of the series.
+        /// </summary>
+        protected internal override void UpdateMaxMin()
+        {
+            if (this.IgnoreExtraplotationForScaling && this.orderedIntervals.Any())
+            {
+                this.MinX = this.Points
+                    .Where(p => !this.orderedIntervals.Any(i => i.Contains(p.X)))
```

Comment:
Probably not a big deal, but it would be nice to use a binary search for this, so that it scales better with large numbers of intervals.

Reply:
Done. I have added an example with 100,000 points and 1000 intervals to the browser.